lib/ruby_indexer/test/constant_test.rb

```diff
@@ -305,6 +307,25 @@ module A
 
       constant = @index["A::M"].first
       assert_instance_of(Entry::Constant, constant)
+
+      # N, O::P and Q::R
+      unresolve_entry = @index["N"].first
+      assert_instance_of(Entry::UnresolvedAlias, unresolve_entry)
+      assert_nil(unresolve_entry.namespace)
+      assert_equal("O::P", unresolve_entry.target)
+
+      resolved_entry = @index.resolve("N", []).first
+      assert_instance_of(Entry::Alias, resolved_entry)
+      assert_equal("O::P", resolved_entry.target)
+
+      # When we are resolving A::I::J, we invoke `resolve("K::L", ["A"])`, which recursively resolves A::K::L too.
+      # Therefore, both A::I::J and A::K::L point to A::M by the end of the previous resolve invocation
+      resolved_entry = @index["O::P"].first
+      assert_instance_of(Entry::Alias, resolved_entry)
+      assert_equal("Q::R", resolved_entry.target)
+
+      constant = @index["Q::R"].first
+      assert_instance_of(Entry::Constant, constant)
```

Comment:
Can we have a new test case for these new assertions? It took me a while to realise they actually don't rely on other parts of the fixture.

Reply:
I don't understand how these are not related. This test case is testing all kinds of ways constant aliases could be defined, but it was never checking for top level constant alias definitions, which were broken.

My changes just add top-level constant alias definition examples to this method as well.